src/Microsoft.Build.Sql/VersionCheckTask.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using Microsoft.Build.Framework;
+using NuGet.Versioning;
+
+namespace Microsoft.Build.Sql;
+
+/// <summary>
+/// This task displays a message in the build log if there is a newer version of the package available on NuGet.org.
+/// </summary>
+public class VersionCheckTask : Microsoft.Build.Utilities.Task, ICancelableTask
+{
+    /// <summary>
+    /// Current version of the package.
+    /// </summary>
+    [Required]
+    public string Version { get; set; } = string.Empty;
+
+    private const string PackageName = "Microsoft.Build.Sql";
+    private CancellationTokenSource _cancellationTokenSource = new();
+
+    public override bool Execute()
+    {
+        if (NuGetVersion.TryParse(Version, out NuGetVersion currentVersion) == false)
+        {
+            Log.LogWarning($"Invalid NuGetVersion format: {Version}");
+            return true;    // This task should not fail build even if the version check fails.
+        }
+
+        try
+        {
+            // Check for prerelease versions if current version is a prerelease version, otherwise check for stable versions only.
+            NuGetVersion latestVersion = NugetClient.GetLatestPackageVersion(
```

Comment:
if using httpclient why do you still need this?

Reply:
ah it's calling the http one. do we have to use NuGetVersion  class? again having dependency on a third party library

Reply:
System.Version doesn't support prerelease versions, alternatively we can use Semver but is also 3rd party. We can also implement our own if you think that's necessary.